source/Calamari.Tests/KubernetesFixtures/HelmVersionParserFixture.cs

```diff
@@ -0,0 +1,34 @@
+using Calamari.Util;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Calamari.Tests.KubernetesFixtures
+{
+    [TestFixture]
+    public class HelmVersionParserFixture
+    {
+        [Test]
+        public void ParseVersion_V2Version_V2()
+        {
+            HelmVersionParser.ParseVersion("Client: v2.16.1+gbbdfe5e").Should().Be(HelmVersion.V2);
+        }
+        
+        [Test]
+        public void ParseVersion_V3Version_V3()
+        {
+            HelmVersionParser.ParseVersion("v3.0.2+g19e47ee").Should().Be(HelmVersion.V3);
+        }
+        
+        [Test]
+        public void ParseVersion_OtherVersion_Null()
+        {
+            HelmVersionParser.ParseVersion("v4.0.2+g19e47ee").Should().BeNull();
+        }
+        
+        [Test]
+        public void ParseVersion_Rubbish_Null()
```

Comment:
do you see any value in testing the input string "V" you have a case in your code to check for it.

Reply:
sure, it's good document the limitations

Reply:
actually scrap that, it might be worth testing uppercase "V3.0.2" and asserting it returns null. I don't think a single "V" makes sense

Reply:
doesn't a single V ensure it enters the return statement of `if (indexOfVersionNumber >= versionString.Length)` not that it matters *that much*?

Reply:
Fair enough, I guess it's worth testing that it doesn't crash 😁